### DISCUSSION TITLE
Pyarmor registration (group license) does not work when upgrading?

### DISCUSSION THREAD

**Discussion by BenjaminYde** [dashingsoft/pyarmor#1439] [General]
## Works
I created a device registration using 8.2.9 using `pyarmor reg -g 1` and then `pyarmor reg -g 1 pyarmor-regfile-xxxx.zip`
Then i get `The device regfile has been generated successfully`, which is ok.

## Breaks
When I upgraded to 8.3.4 and used the same cli commands in the terminal on the host, it does not work anymore because the `pyarmor-group-device.1` is different?

I get the following error with 8.3.4:
`ERROR device (1) is not for this machine`

**8.2.9:**
```
machine: xxxx
```

vs 

**8.3.4:**
```
# Generated by Pyarmor 8.3.4, xxxx
host: xxx
system: Linux (xxx)
machine: xxxx
```

Why is this not backwards compatible?
I should be able to upgrade my pyarmor version with the same license...

**Comment by jondy** [comment]
It should be backwards compatible.

Check the log with debug option to register Pyarmor `pyarmor -d reg pyarmor-regfile-group-device-xxx.zip`

If even the old machine is mismatched, please re-generate device file by Pyarmor 8.3.4. Because old machine id in some machines will be changed after reboot, that's why new machine id is introduced.

**Comment by BenjaminYde** [comment]
On the host machine I want to generate my device file (with id 1) using 8.3.4 and the following commands:
```
pyarmor reg -g 1
pyarmor reg -g 1 pyarmor-regfile-xxxx.zip
``` 
The first command is OK, but the second command gives me the following error:
```
INFO     Python 3.10.12
INFO     Pyarmor 8.3.4 (trial), 000000, non-profits
INFO     Platform linux.x86_64
INFO     register device file ".pyarmor/group/pyarmor-group-device.1"
INFO     use group license "pyarmor-regfile-6054.zip"
INFO     send request to server
ERROR    device (1) is not for this machine
```
The same commands with 8.2.9 work just fine.

When I use the `-d` flag, I just get a long url `url: https://api.dashingsoft.com/product/key/enter/group/xxxx` and the same error.

Pyarmor 8.2.9 and Pyarmor 8.3.4 generate both different machine id's. Shouldn't this be the same id?

So when I reboot it is possible that I need to use a new machine id? I will lose the other one??
Why is this not the same id?

**Reply by BenjaminYde** [reply]
It works when using a new device id, but this is not what I want...
The device id keeps resetting/changing when the deivice reboots? 
This means when I reboot 100 times, all my device id's will be non-usable anymore?

**Comment by jondy** [comment]
Generally the machine id will be same in same machine, and that's why introduce new machine id in v8.3.x, which remove bridge network interface in order to make machine id keep same.

But the old machine id (generated in the v8.2.9) should also work if no any system settings are changed, I'll check this backward compatible.
